### PR TITLE
[INLONG-10823][SDK] Transform SQL support Reverse function

### LINKED ISSUE #10823: [Feature][SDK] Transform SQL support Reverse function
### Description

- parent  #10796
- REVERSE(str)
Returns the string str with the order of the characters reversed.
- Such as: REVERSE("abc") = "cba"

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ying-hua**
Fixes #10823

### Motivation

Add one transform function class: ReverseFunction and its unit test.

 - REVERSE(str)
    Returns the string str with the order of the characters reversed.
 - Such as: REVERSE("abc") = "cba"

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by aloyszhang — APPROVED** [review]

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+     * @param sourceData
+     * @param rowIndex
+     * @param context
+     * @return
+     */
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object stringObj = stringParser.parse(sourceData, rowIndex, context);
+        String str = OperatorTools.parseString(stringObj);
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java:50]: According to the FlinkSQL document, when the parameter `string` is NULL, it will return NULL. But if `stringObj` is NULL, it will cause `NullPointerException` here. You should check if `stringObj` is NULL before parse it.
image

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformStringFunctionsProcessor.java [hunk 2]
```diff
@@ -184,4 +186,29 @@ public void testTrimFunction() throws Exception {
...
+                .create(config2, SourceDecoderFactory.createCsvDecoder(csvSource),
+                        SinkEncoderFactory.createKvEncoder(kvSink));
+        List<String> output2 = processor2.transform("apple|ban ana |cloud|2|1|3", new HashMap<>());
+        Assert.assertEquals(1, output2.size());
+        Assert.assertEquals(output2.get(0), "result= ana nab");
+        // case3: reverse(12345)
+        List<String> output3 = processor1.transform("12345|banana|cloud|2|1|3", new HashMap<>());
+        Assert.assertEquals(1, output3.size());
+        Assert.assertEquals(output3.get(0), "result=54321");
+    }
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformStringFunctionsProcessor.java:213]: It is better to add a test case that the parameter is NULL.

Comment (ying-hua) [reply inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformStringFunctionsProcessor.java:217]: done

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java (hunk 1 above)
Comment (ying-hua) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/ReverseFunction.java:53]: done

**Comment by dockerzhang** [comment]
@ying-hua please fix the conflicts, thanks.

**Comment by ying-hua** [comment]
> @ying-hua please fix the conflicts, thanks.

done

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformStringFunctionsProcessor.java [hunk 3]
```diff
@@ -184,6 +186,36 @@ public void testTrimFunction() throws Exception {
...
+                .create(config2, SourceDecoderFactory.createCsvDecoder(csvSource),
+                        SinkEncoderFactory.createKvEncoder(kvSink));
+        List<String> output2 = processor2.transform("apple|ban ana |cloud|2|1|3", new HashMap<>());
+        Assert.assertEquals(1, output2.size());
+        Assert.assertEquals(output2.get(0), "result= ana nab");
+        // case3: reverse(12345)
+        List<String> output3 = processor1.transform("12345|banana|cloud|2|1|3", new HashMap<>());
+        Assert.assertEquals(1, output3.size());
+        Assert.assertEquals(output3.get(0), "result=54321");
+        // 4: reverse(null)
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformStringFunctionsProcessor.java:213]:
```suggestion
        // case4: reverse(null)
```

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Review by yfsn666 — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]
